components/popconfirm/popconfirm.spec.ts

```diff
@@ -111,6 +111,20 @@ describe('NzPopconfirm', () => {
     expect(component.confirm).toHaveBeenCalledTimes(1);
     expect(component.cancel).toHaveBeenCalledTimes(0);
   }));
+
+  it('should support changing visibility programmatically', fakeAsync(() => {
```

Comment:
Shared logic among tooltip, popover and popconfirm should be tested in tooltip for only once.